bootstrap/app.go

```diff
@@ -17,9 +19,42 @@ var grpcServer *grpc.Server
 var httpServer *http.Server
 var waitForCompletion *sync.WaitGroup
 
+// Cfg holds the parsed and validated service config.
+var Cfg *serviceconfig.ServiceConfig
+
+// getConfig loads the config based on the technical implementation "viper".
+func getConfig(configPath string) (serviceconfig.ServiceConfig, error) {
+	cfg, err := NewConfigurationBuilder().
+		ConfigFilePath(configPath).
+		NoDefaults().
+		Build()
+
+	if err != nil {
+		glog.Fatalf("Could not initialize config: %v", err)
+	}
+
+	return cfg.Load()
+}
+
 // Run configures and runs the actual bootstrap.
-func Run(endpoint string, oidcDiscoveryEndpoint string, token string, store string, useTLS bool) {
-	grpcServer, httpServer = initialize(endpoint, oidcDiscoveryEndpoint, token, store, useTLS)
+func Run(configPath string) {
+	srvCfg, err := getConfig(configPath)
+	if err != nil {
+		glog.Fatal("Unable to load configuration: %v", err)
+	}
+	vl := validator.New()
+	err = vl.Struct(srvCfg)
+
+	if err != nil {
+		for _, e := range err.(validator.ValidationErrors) {
+			glog.Errorf("Error in configuration: %v", e)
+		}
+		glog.Fatal("Can not start service with wrong configuration.")
+	}
+	//set global Config now that it is parsed and validated.
+	Cfg = &srvCfg
```

Comment:
Hmm... no problem with global config, but shouldn't we then access config globally and not pass srvCfg everywhere as an argument?

I suggest we remove this until we are ready/willing to use it.

Reply:
This is a good point. One of the items on Dominik's big list of suggestions was to see if we could remove that- we probably can.